Source/DafnyServer/boogie/DafnyProvider.cs

```diff
@@ -298,25 +267,23 @@ Model.Element Unbox(Model.Element elt)
       var unboxed = f_box.AppWithResult(elt);
       if (unboxed != null)
         return unboxed.Args[0];
-      else
-        return elt;
+      return elt;
     }
   }
 
-  public class StateNode : NamedState
+  public class DafnyModelState
   {
     internal readonly DafnyModel dm;
-    public readonly List<VariableNode> Vars = new List<VariableNode>();
-    internal readonly List<VariableNode> skolems;
-    internal readonly int index;
+    public readonly List<VariableNode> Vars = new();
+    private readonly List<VariableNode> skolems;
+    private Model.CapturedState state;
+    private LanguageModel langModel; // no point making it protected - they will need VccModel, DafnyModel
```

Comment:
I find the comment confusing.

Reply:
Thank you @keyboardDrummer ! I've removed it. The comment wasn't mine - accidentally copied it when moving this line from [here](https://github.com/dafny-lang/dafny/blob/8b26545c041cdfa7bdf39381277e1b11a745903c/Source/DafnyServer/boogie/Namer.cs#L490).

Reply:
I figured as much ^^